src/puppetlabs/puppetdb/utils.clj

```diff
@@ -232,3 +237,25 @@
   (reduce-kv (fn [acc k v]
                (assoc acc (schema.core/required-key (puppetlabs.puppetdb.utils/kwd->str k)) v))
              {} kwd-schema))
+
+(defn as-path
+  "Create a url path from arguments. Does not append a slash to the beginning
+   or end. Example:
+   (as-path '/v4' 'facts' "
+  [root & path]
+  (apply str root "/" (string/join "/" path)))
+
+(pls/defn-validated child->expansion
+  "Convert child to the expanded format."
+  [data :- (s/maybe (s/either PGobject s/Str))
+   parent :- s/Keyword
+   child :- s/Keyword
+   base-url :- s/Str]
```

Comment:
this should probably be `base-url-str` of `base-path` something to distinguish it from the map called `base-url` we use elsewhere